module/validation/seal_validator.go

```diff
@@ -44,26 +44,27 @@ func NewSealValidator(
 	results storage.ExecutionResults,
 	seals storage.Seals,
 	assigner module.ChunkAssigner,
-	requiredApprovalsForSealConstruction uint,
+	requiredApprovalsForSealConstructionGetter module.RequiredApprovalsForSealConstructionInstanceGetter,
 	requiredApprovalsForSealVerification uint,
 	metrics module.ConsensusMetrics,
 ) (*sealValidator, error) {
+	requiredApprovalsForSealConstruction := requiredApprovalsForSealConstructionGetter.GetValue()
 	if requiredApprovalsForSealConstruction < requiredApprovalsForSealVerification {
```

Comment:
I feel like this getter approach is a bit dangerous since we basically render this check useless.

Reply:
Validation is added to ValidateRequireApprovals, which got run in `SetValue`